src/conformance/ValidationService.ts

```diff
@@ -60,7 +63,120 @@ export class ValidationService {
             this.validators.setValue(type, relevantSet);
         });
     }
+}
+
+/**
+ * A buffer containing pending tasks that are in need of completion.
+ * This class serializes the completions of these pending tasks, meaning that the completions
+ * will be performed in the same order that the pending tasks themselves entered the buffer.
+ */
+export class TaskCompletionBuffer {
+
+    private queue: Collections.PriorityQueue<PendingTask>;
+
+    /**
+     * Create a new TaskCompletionBuffer.
+     */
+    public constructor() {
+        this.queue = new Collections.PriorityQueue<PendingTask>(function(a: PendingTask, b: PendingTask) {
+            // having a "larger" timestamp means having a lower priority
+            if (a.getTimeStamp() > b.getTimeStamp()) {
```

Comment:
I think this somewhat complex function can be more concisely expressed as

```
return Collections.util.reverseCompareFunction(Collections.util.defaultCompare)(
    a.getTimeStamp(),
    b.getTimeStamp());
```

(I haven't checked though: you might want to try it out first.)

Reply:
Or just `return -Collections.util.defaultCompare(a.getTimeStamp(), b.getTimeStamp());`. That doesn't really express your intent quite as much as the `reverseCompareFunction` call, though.